api/client/build.go

```diff
@@ -265,6 +268,14 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	headers.Add("X-Registry-Config", base64.URLEncoding.EncodeToString(buf))
 	headers.Set("Content-Type", "application/tar")
 
+	// collect all the build-time environment variables for the container
```

Comment:
I think you are asking if a user were to specify same arg twice `--build-arg` then if behavior  is deterministic based on order of specification?

I think it may not be true. As it depends on the order in which `github.com/docker/docker/cli` package parses the passed flags and it's implementation which I haven't looked at. 

Does it helps if we document that behavior it is in-deterministic or might change as it is dependent on the package we use to parse flags?